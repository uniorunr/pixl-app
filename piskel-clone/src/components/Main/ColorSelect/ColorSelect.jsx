import React, { Component } from 'react';
import './ColorSelect.scss';
import PropTypes from 'prop-types';

class ColorSelect extends Component {
  constructor(props) {
    super(props);
    this.primary = React.createRef();
    this.secondary = React.createRef();
  }

  handlePrimaryChange = ({ target }) => {
    const { updateColor } = this.props;
    if (target === this.primary.current) {
      updateColor(target.value, 'primary');
      sessionStorage.setItem('primaryColor', target.value);
    } else {
      updateColor(target.value);
      sessionStorage.setItem('secondaryColor', target.value);
    }
  };

  handleSwap = () => {
    const { updateColor, primaryColor, secondaryColor } = this.props;
    updateColor(secondaryColor, 'primary');
    sessionStorage.setItem('primaryColor', secondaryColor);
    updateColor(primaryColor);
    sessionStorage.setItem('secondaryColor', primaryColor);
  };

  render() {
    const { primaryColor, secondaryColor } = this.props;

    return (
      <section className="color-select-section">
        <div className="colors">
          <div>
            <input
              type="color"
              className="colors__input-primary"
              value={primaryColor}
              ref={this.primary}
              onChange={this.handlePrimaryChange}
            />
            <div
              className="colors__primary"
              style={{ backgroundColor: primaryColor }}
            />
          </div>
          <div>
            <input
              type="color"
              className="colors__input-secondary"
              value={secondaryColor}
              ref={this.secondary}
              onChange={this.handlePrimaryChange}
            />
            <div
              className="colors__secondary"
              style={{ backgroundColor: secondaryColor }}
            />
          </div>
          <button
            type="button"
            className="colors__swap"
            onClick={this.handleSwap}
          >
            swap
          </button>
        </div>
      </section>
    );
  }
}

ColorSelect.propTypes = {
  updateColor: PropTypes.func.isRequired,
  primaryColor: PropTypes.string.isRequired,
  secondaryColor: PropTypes.string.isRequired,
};

export default ColorSelect;
