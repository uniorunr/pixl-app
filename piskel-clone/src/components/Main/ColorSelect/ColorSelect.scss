.color-select-section {
  .colors {
    display: flex;
    position: relative;
    border: 2px solid black;
    margin: 20px 10px 0 20px;
    padding: 10px;
    width: 45px;
    height: 40px;

    .colors__input-primary {
      z-index: 3;
      position: absolute;
      opacity: 0;
      width: 30px;
      height: 30px;
      top: 5px;
      left: 5px;
      cursor: pointer;
    }

    .colors__input-secondary {
      z-index: 1;
      position: absolute;
      opacity: 0;
      width: 30px;
      height: 30px;
      top: 20px;
      right: 5px;
      cursor: pointer;
    }

    .colors__primary {
      width: 30px;
      height: 30px;
      border: 2px solid black;
      position: absolute;
      top: 5px;
      left: 5px;
      z-index: 2;
      cursor: pointer;
    }

    .colors__secondary {
      width: 30px;
      height: 30px;
      border: 2px solid black;
      position: absolute;
      top: 20px;
      right: 5px;
      cursor: pointer;
    }
  }
}
