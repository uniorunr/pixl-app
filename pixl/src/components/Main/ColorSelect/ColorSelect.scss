.color-select-section {
  .colors {
    display: flex;
    position: relative;
    border: 2px solid black;
    margin: 15px auto;
    padding: 15px;
    width: 45px;
    height: 40px;

    .colors__input-primary {
      z-index: 3;
      position: absolute;
      opacity: 0;
      width: 30px;
      height: 30px;
      top: 10px;
      left: 10px;
      cursor: pointer;
    }

    .colors__input-secondary {
      z-index: 1;
      position: absolute;
      opacity: 0;
      width: 30px;
      height: 30px;
      top: 25px;
      right: 10px;
      cursor: pointer;
    }

    .colors__primary {
      width: 30px;
      height: 30px;
      border: 2px solid black;
      position: absolute;
      top: 10px;
      left: 10px;
      z-index: 2;
      cursor: pointer;
    }

    .colors__secondary {
      width: 30px;
      height: 30px;
      border: 2px solid black;
      position: absolute;
      top: 25px;
      right: 10px;
      cursor: pointer;
    }

    .colors__swap {
      position: absolute;
      bottom: 5px;
      left: 5px;
      background: transparent;
      border: none;
      outline: none;
      cursor: pointer;
    }
  }
}
