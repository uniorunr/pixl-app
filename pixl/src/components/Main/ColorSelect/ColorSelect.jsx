import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from '../../../actions/actions';
import './ColorSelect.scss';

class ColorSelect extends Component {
  primary = React.createRef();

  secondary = React.createRef();

  handlePrimaryChange = ({ target }) => {
    const { updateColor } = this.props;
    if (target === this.primary.current) {
      updateColor(target.value, 'primary');
      sessionStorage.setItem('primaryColor', target.value);
    } else {
      updateColor(target.value);
      sessionStorage.setItem('secondaryColor', target.value);
    }
  };

  handleSwap = () => {
    const { updateColor, primaryColor, secondaryColor } = this.props;
    updateColor(secondaryColor, 'primary');
    sessionStorage.setItem('primaryColor', secondaryColor);
    updateColor(primaryColor);
    sessionStorage.setItem('secondaryColor', primaryColor);
  };

  render() {
    const { primaryColor, secondaryColor } = this.props;

    return (
      <section className="color-select-section">
        <div className="colors">
          <div>
            <input
              type="color"
              className="colors__input-primary"
              value={primaryColor}
              ref={this.primary}
              onChange={this.handlePrimaryChange}
            />
            <div
              className="colors__primary"
              style={{ backgroundColor: primaryColor }}
            />
          </div>
          <div>
            <input
              type="color"
              className="colors__input-secondary"
              value={secondaryColor}
              ref={this.secondary}
              onChange={this.handlePrimaryChange}
            />
            <div
              className="colors__secondary"
              style={{ backgroundColor: secondaryColor }}
            />
          </div>
          <button
            type="button"
            className="colors__swap"
            onClick={this.handleSwap}
          >
            <i className="fas fa-sync-alt" />
          </button>
        </div>
      </section>
    );
  }
}

ColorSelect.propTypes = {
  updateColor: PropTypes.func.isRequired,
  primaryColor: PropTypes.string.isRequired,
  secondaryColor: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  primaryColor: state.colors.primaryColor,
  secondaryColor: state.colors.secondaryColor,
});

const mapDispatchToProps = (dispatch) => {
  const { updateColor } = bindActionCreators(actions, dispatch);
  return {
    updateColor: (color, isPrimary) => {
      updateColor(color, isPrimary);
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ColorSelect);
