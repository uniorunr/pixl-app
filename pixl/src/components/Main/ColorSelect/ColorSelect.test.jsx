import React from 'react';
import renderer from 'react-test-renderer';
import ColorSelect from './ColorSelect';

function createNodeMock(element) {
  if (element.type === 'input') {
    return {};
  }
  return null;
}

it('renders correctly', () => {
  const options = { createNodeMock };

  const tree = renderer
    .create(
      <ColorSelect
        updateColor={() => {}}
        primaryColor="#000000"
        secondaryColor="#ffffff"
      />,
      options,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
