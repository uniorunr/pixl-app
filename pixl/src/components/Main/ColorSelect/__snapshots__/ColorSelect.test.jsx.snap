// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="color-select-section"
>
  <div
    className="colors"
  >
    <div>
      <input
        className="colors__input-primary"
        onChange={[Function]}
        type="color"
        value="#000000"
      />
      <div
        className="colors__primary"
        style={
          Object {
            "backgroundColor": "#000000",
          }
        }
      />
    </div>
    <div>
      <input
        className="colors__input-secondary"
        onChange={[Function]}
        type="color"
        value="#ffffff"
      />
      <div
        className="colors__secondary"
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      />
    </div>
    <button
      className="colors__swap"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-sync-alt"
      />
    </button>
  </div>
</section>
`;
